pkg/sql/opt/memo/statistics_builder.go

```diff
@@ -863,9 +864,20 @@ func (sb *statisticsBuilder) constrainScan(
 				// estimate 0 rows.)
 				s.RowCount = max(inputHist.ValuesCount(), 1)
 				if colStat, ok := s.ColStats.Lookup(colSet); ok {
-					colStat.Histogram = inputHist.InvertedFilter(scan.InvertedConstraint)
-					histCols.Add(invertedConstrainedCol)
-					sb.updateDistinctCountFromHistogram(colStat, inputStat.DistinctCount)
+					// If we have a histogram that doesn't contain DBytes, we know we're
+					// looking at a forward histogram, since inverted histograms always
+					// are of type DBytes. We cannot use the histogram if this is the
+					// case.
+					// Note that this is a sketchy situation (no pun intended), because
```

Comment:
Not only is it sketchy, it means that we won't have accurate row count estimates for queries on trigram indexes. As a result, the optimizer probably won't pick the best plan in a number of cases.

Reply:
Yeah :( And even worse, isn't it actually more like we won't have accurate row count estimates for queries against text columns that happen to also be inverted indexed? I think that's the thing I'm most scared of here...

Reply:
Correct, that's what I meant to say.